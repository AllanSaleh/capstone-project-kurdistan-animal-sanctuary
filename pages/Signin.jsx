import React from "react";
import { useTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import Navbar from "../components/Navbar";
import Signin from "../components/signin";
import Footer from "../components/Footer";

export default function Home() {
  // let name = 'Allan';
  const { t } = useTranslation("signIn");
  return (
    <>
      <head>
        <title>{t("signIn.title")}</title>
      </head>
      <Navbar />

      <Signin />

      <Footer />
    </>
  );
}

export async function getStaticProps({ locale }) {
  return {
    props: {
      ...(await serverSideTranslations(locale, [
        "common",
        "navbar",
        "footer",
        "signIn",
      ])),
      // Will be passed to the page component as props
    },
  };
}
