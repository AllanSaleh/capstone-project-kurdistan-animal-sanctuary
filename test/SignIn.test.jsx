import React from "react";
import renderer from "react-test-renderer";
import { test, expect } from "@jest/globals";
import Signin from "../components/signin";

test("proparly renders the Signin component", () => {
  const tree = renderer.create(<Signin />).toJSON();
  expect(tree).toMatchSnapshot();
});
