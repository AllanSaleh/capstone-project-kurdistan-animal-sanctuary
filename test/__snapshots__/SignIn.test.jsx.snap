// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proparly renders the Signin component 1`] = `
<div
  className="bg-background h-screen bg-clip-content bg-no-repeat  bg-cover flex"
>
  <backgroundImage />
  <div
    className="m-auto bg-secondary bg-opacity-80 max-w-lg rounded-xl text-center w-80 md:w-96"
  >
    <div
      className="w-60 md:w-80 m-auto "
    >
      <form
        onSubmit={[Function]}
      >
        <p
          className="text-primary text-4xl sm:text-5xl text-center font-Quicksand p-10 font-bold"
        >
          signIn.signIn
        </p>
        <div
          className="flex flex-col "
        >
          <input
            className="border mb-6 border-primary rounded text-primary text-sm h-10 bg-secondary"
            faIcon="envelope"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="signIn.email"
            type="email"
            value=""
          />
          <input
            className="border  mb-6 mx-auto sm:mx-0 w-4/5 sm:w-full border-primary rounded  text-primary text-sm h-10 bg-secondary"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            placeholder="signIn.password"
            type="password"
            value=""
          />
        </div>
        <label
          className="text-primary"
          htmlFor="membrme"
        >
          <input
            className="mb-4"
            name="membrme"
            type="checkbox"
          />
          signIn.rememberMe
        </label>
        <button
          className="place-items-center text rounded text-center mb-6 py-2 px-20 font-bold text-primary bg-highlight hover:bg-darken"
          type="submit"
        >
          signIn.signIn
        </button>
      </form>
    </div>
    <div
      className="text-center "
    >
      <div
        className="flex justify-evenly "
      >
        <span>
          <a
            href="http://www.facebook.com"
          >
            <img
              alt="fb"
              className="w-10 "
              src="https://seeklogo.com/images/F/facebook-icon-circle-logo-09F32F61FF-seeklogo.com.png"
            />
          </a>
        </span>
        <span>
          <a
            href="https://twitter.com/"
          >
            <img
              alt="twitter"
              className="w-10 "
              src="https://cdn4.iconfinder.com/data/icons/social-media-icons-the-circle-set/48/twitter_circle-512.png"
            />
          </a>
        </span>
        <span>
          <a
            href="http://google.com"
          >
            <img
              alt="Google"
              className="w-10"
              src="https://www.freepnglogos.com/uploads/google-logo-png/google-logo-icon-png-transparent-background-osteopathy-16.png"
            />
          </a>
        </span>
      </div>
      <div
        className="py-5"
      >
        <a
          href="forgetpassword"
        >
          signIn.forgetPassword
        </a>
         
        <br />
        <a
          href="signup"
        >
          signIn.dontHaveAccount
        </a>
      </div>
    </div>
  </div>
</div>
`;
